src/lab11.py

```diff
@@ -417,28 +419,43 @@ def __init__(self):
             self.BLUE_MASK = 0x00ff0000
             self.ALPHA_MASK = 0xff000000
 
+    def init_chrome(self):
+        self.chrome_font = get_font(20, "normal", "roman")
+        chrome_font_height = linespace(self.chrome_font)
+
+        self.padding = 5
+        self.tab_header_bottom = chrome_font_height + 2 * self.padding
+        self.addressbar_top = self.tab_header_bottom + self.padding
+        self.chrome_bottom = \
+            math.ceil(
+                self.addressbar_top + chrome_font_height + 2 * self.padding)
+
     def handle_down(self):
         self.tabs[self.active_tab].scrolldown()
         self.draw()
 
     def handle_click(self, e):
-        if e.y < CHROME_PX:
+        if e.y < self.chrome_bottom:
             self.focus = None
-            if 40 <= e.x < 40 + 80 * len(self.tabs) and 0 <= e.y < 40:
-                self.active_tab = int((e.x - 40) / 80)
-                self.tabs[self.active_tab].render()
-                self.raster_tab()
-            elif 10 <= e.x < 30 and 10 <= e.y < 30:
+            if intersects(e.x, e.y, self.plus_bounds()):
```

Comment:
Oh man, this code is kind of ugly. I hadn't thought about hit testing.

Reply:
Really? It seems much more readable to me than the old code.